csrc/include/sampling.cuh

```diff
@@ -0,0 +1,1431 @@
+// SPDX-License-Identifier: MIT
```

Comment:
These files are coming from flashinfer, we use Apache License 2.
You must follow https://github.com/flashinfer-ai/flashinfer/blob/8d29762c8780c9065da50bede23b91b861c413c9/LICENSE#L89-L121
to redistribute the code and keep the original copyright file and notices.